### DISCUSSION TITLE
Stylus for mobile ?

### DISCUSSION THREAD

**Discussion by giias** [openstyles/stylus#1197] [Q&A]
Any idea when/if Stylus will be available on mobile again like it used to ? I used to use it on Firefox Mobile and I really liked it

I know Mozilla made changes to how add-ons work on their new Firefox Mobile some time ago, and they've been adding more support for add-ons. Actually I just found earlier today that there's a way to still use Stylus on Android/Firefox mobile and it has been working well enough for me so far... but currently the only way I know for mobile/Android is through Firefox Nightly by using the "add-on collection" feature as shown here-

https://www.reddit.com/r/firefox/comments/j1ta1s/you_can_now_add_any_addon_in_firefox_nightly/
https://blog.mozilla.org/addons/2020/09/29/expanded-extension-support-in-firefox-for-android-nightly/

thing is Firefox Nightly is basically a beta/alpha browser, and there are people who prefer the main/regular Firefox mobile for stability etc... so it'd be nice if Stylus could work again on the main release for Firefox mobile on Android like it used to. Currently if I check the add-ons page on the main Firefox mobile, there are only 18 recommended add-ons you can select to add and Stylus is not included (also, no option to use the "add-on collection" feature to add new/more add-ons there since apparently that's only for Firefox Nightly). Stylus is a recommended add-on for Firefox on desktop and being updated constantly... any idea what it would take to have it added as a recommended and available add-on for the main release of Firefox mobile ? Anyway, thanks for the work on Stylus! It's definitely one of my favorite add-ons -- just wish I and others could use it more easily on mobile with main Firefox mobile like in the past...

**Comment by tophf** [comment]
@eight04 is the only one I know who might be interested to have a look into this but he's super busy so don't get your hopes up.

**Reply by Mottie** [reply]
IIRC, there is an option to make it available for Firefox mobile when creating a new version on AMO. I just haven't checked it because I wasn't sure if we supported mobile. I'll look for it next update.

**Reply by tophf** [reply]
Stylus worked in mobile Firefox for quite a long time a few years ago. I don't know what happened since, probably AMO/FF broke something or we just dropped the ball because no one cared.

**Reply by narcolepticinsomniac** [reply]
We started failing automated reviews around the same time sync was implemented. Best guess was that it was due to `launchWebAuthFlow` not being supported in FF mobile. Since sync was rewritten recently, I kinda figured it might get re-approved for mobile again, but apparently we also need to opt back in.

I don't use it on mobile personally, but users reported they were able to install from the regular AMO [versions page](https://addons.mozilla.org/en-US/firefox/addon/styl-us/versions/). Sync wouldn't work previously, but I guess it might now. Supposedly, everything else worked decently.

**Comment by eight04** [comment]
> any idea what it would take to have it added as a recommended and available add-on for the main release of Firefox mobile ?

https://addons.mozilla.org/en-US/firefox/collections/4757633/7dfae8669acc4312a65e8ba5553036/

I think they were picked manually by Mozilla. The last update date is Oct 14, 2020.

**Reply by harry-wg** [reply]
Could we, as a stopgap measure, have a cut down "execute only" version that can import an existing user style from desktop stylus and somehow apply it to to the pages of the matching domain?

**Comment by 333hronos** [comment]
By the way, fully supports almost all Chrome extensions (including Stylus) Yandex browser.

**Comment by giias** [comment]
Huh I don't remember seeing the Yandex mobile browser having support for extensions last time I tried it a year (or 2 ?) ago... but I just tested it again on my android phone, and yep Stylus seems to be working fine on the Yandex mobile browser - I was able to change the css/colors just fine for a mobile site I tried.

I also checked to see if Stylus had finally been added to the list of available addons on Firefox mobile for android, and I found that there's a bigger list of addons available compared to last time I checked but sadly Stylus is still not available in Firefox Mobile (it can still be added and used in Firefox Nighty/beta mobile through the collections addon feature, but it's still not available officially in the list of addons for the main/stable Firefox mobile). Yandex is a stable mobile browser I guess and supports Stylus, but honestly I'm not a fan of Yandex's UI, and I'm used to using Firefox...

I wonder if there's a place where we can request or ask Mozilla to add Stylus if they're adding this manually. Stylus has worked well on mobile in the past and it's even a Firefox recommended addon for the desktop version of Firefox. Not sure why Mozilla still hasn't officially added it as a recommended addon for the mobile browser also. By the way, thanks for the replies
